app/models/metric.rb

```diff
@@ -3,7 +3,21 @@ class Metric < ApplicationRecord
 
   include Metric::Common
 
-  def self.reindex_table_name
-    "metrics_#{Time.now.utc.hour + 1}"
+  def self.table_name
+    Thread.current[:metric_table] || super
+  end
+
+  def self.with_table_for_time(hour = Time.now.utc.hour)
+    Thread.current[:metric_table] = reindex_table_name(hour)
+    yield
+    self
+  ensure
+    Thread.current[:metric_table] = nil
+  end
+
+  # reindex_table_name # tablename for "now"
+  # reindex_Table_name(4) # tablename for hour number 4
```

Comment:
These comments barely make sense to me, and only after I understood everything that was going on around them could I figure out what the purpose of this was.

Can these be written `yard` format or something more standard?